packages/snjs/lib/application.ts

```diff
@@ -554,6 +554,14 @@ export class SNApplication {
     return compareVersions(userVersion, ProtocolVersion.V004) >= 0;
   }
 
+  public getUserSubscription(): Promise<HttpResponse | GetSubscriptionResponse> {
+    return this.sessionManager.getSubscription();
+  }
+
+  public getSubscriptions(): Promise<HttpResponse | GetSubscriptionsResponse> {
+    return this.apiService.getSubscriptions();
+  }
+
```

Comment:
Agreed

Reply:
Isn't it possible that we could want for example the error message from the error response to get to the frontend? Also checking, it seems we're already using this pattern on the sessions methods exposed on application

Reply:
I've changed it 🙂

Reply:
> Isn't it possible that we could want for example the error message from the error response to get to the frontend? Also checking, it seems we're already using this pattern on the sessions methods exposed on application

The common pattern is that we put the error in the Promise. Unfortunately, Typescript doesn't allow type inference for Errors in the Promises so we have to use the base form. But that's fine in this case since you don't want to process statuses in the view layer, you just trigger events and show the message once processed. The drawback of that approach comes when you want to support multiple languages, but the error messages are the least of the concerns since they're easy to discover - just search for ".catch" and "Error(.*)".